ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AbstractBlockParameterMethod.java

```diff
@@ -57,21 +60,44 @@ protected Object latestResult(final JsonRpcRequestContext request) {
     return resultByBlockNumber(request, blockchainQueriesSupplier.get().headBlockNumber());
   }
 
+  protected Object finalizedResult(final JsonRpcRequestContext request) {
+    return posRelatedResult(request, BlockchainQueries::finalizedBlockHeader);
+  }
+
+  protected Object safeResult(final JsonRpcRequestContext request) {
+    return posRelatedResult(request, BlockchainQueries::safeBlockHeader);
+  }
+
+  private Object posRelatedResult(
+      final JsonRpcRequestContext request,
+      final Function<BlockchainQueries, Optional<BlockHeader>> blockHeaderSupplier) {
+
+    return blockHeaderSupplier
+        .apply(blockchainQueriesSupplier.get())
+        .map(header -> resultByBlockNumber(request, header.getNumber()))
+        .orElseGet(
+            () ->
+                new JsonRpcErrorResponse(request.getRequest().getId(), JsonRpcError.UNKNOWN_BLOCK));
+  }
+
   protected Object findResultByParamType(final JsonRpcRequestContext request) {
     final BlockParameter blockParam = blockParameter(request);
 
-    final Object result;
     final Optional<Long> blockNumber = blockParam.getNumber();
     if (blockNumber.isPresent()) {
-      result = resultByBlockNumber(request, blockNumber.get());
-    } else if (blockParam.isLatest()) {
-      result = latestResult(request);
-    } else {
-      // If block parameter is not numeric or latest, it is pending.
-      result = pendingResult(request);
+      return resultByBlockNumber(request, blockNumber.get());
+    }
```

Comment:
Style wise these should remain if/else chained.  They are not proper "guard blocks" and readability would be improved by retaining the if/elseif/elseif chain.  Bringing in the last return as an else would further improve the readability.

Reply:
done